kaldb/src/test/java/com/slack/kaldb/clusterManager/ReplicaCreatorServiceTest.java

```diff
@@ -0,0 +1,164 @@
+package com.slack.kaldb.clusterManager;
+
+import static com.slack.kaldb.config.KaldbConfig.DEFAULT_START_STOP_DURATION;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import brave.Tracing;
+import com.slack.kaldb.metadata.replica.ReplicaMetadata;
+import com.slack.kaldb.metadata.replica.ReplicaMetadataStore;
+import com.slack.kaldb.metadata.snapshot.SnapshotMetadata;
+import com.slack.kaldb.metadata.snapshot.SnapshotMetadataStore;
+import com.slack.kaldb.proto.config.KaldbConfigs;
+import com.slack.kaldb.server.MetadataStoreService;
+import com.slack.kaldb.testlib.MetricsUtil;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Objects;
+import java.util.UUID;
+import org.apache.curator.test.TestingServer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ReplicaCreatorServiceTest {
+
+  private static final int REPLICAS_PER_SNAPSHOT_TEST = 2;
+
+  private TestingServer testingServer;
+  private MeterRegistry meterRegistry;
+
+  @Before
+  public void setup() throws Exception {
+    Tracing.newBuilder().build();
+    meterRegistry = new SimpleMeterRegistry();
+    testingServer = new TestingServer();
+  }
+
+  @After
+  public void shutdown() throws IOException {
+    testingServer.close();
+    meterRegistry.close();
+  }
+
```

Comment:
We should also add unit tests when stores are empty and when first snapshot is added. 

Also, we should also add some unit tests to make sure that this code is robust against ZK failures. For example, if we fail to create a replica, the next run of the program should create it.

Reply:
Done.